goon/browserassets/js/browserOutput.js

```diff
@@ -871,32 +871,73 @@ $(function() {
 		setCookie('pingdisabled', (opts.pingDisabled ? 'true' : 'false'), 365);
 	});
 
-	$('#saveLog').click(function(e) {
-		var saved = '';
+	$('#saveLog').click(function(e) {	//note that this will only work in browsers supporting CORS (IE 8+ and literally every alternative)
+		var makeHaste = function(txt) { //creates a haste and returns the url
+			if (window.XMLHttpRequest) {
+				hasteRequest = new XMLHttpRequest();
+			} else {
+				hasteRequest = new ActiveXObject("Microsoft.XMLHTTP");
+			}
+			hasteRequest.open('POST', 'http://localhost:7777/documents', true);
```

Comment:
Could you make the `localhost:7777` part of this a setting in `opts`? https://github.com/ParadiseSS13/Paradise/blob/master/goon/browserassets/js/browserOutput.js#L24